libs/utils/analysis/frequency_analysis.py

```diff
@@ -110,6 +110,69 @@ def _dfg_cluster_frequency_residency(self, cluster, total=True):
             return residency.total
         return residency.active
 
+    def _dfg_frequency_transitions(self, entity):
+        """
+        Compute number of frequency transitions of a given entity. Entity can
+        be a CPU or a cluster name. Notice that we assume that frequency is
+        scaled at cluster level, therefore we always consider the first CPU of
+        a cluster for this computation.
+
+        Requires cpu_frequency events to be available in the trace.
+
+        :param entity: a CPU ID or a cluster name
+        :type entity: int or str
+
+        :returns: :mod:`pandas.DataFrame` - number of frequency transitions
+        """
+        if not self._trace.hasEvents('cpu_frequency'):
+            self._log.warn('Events [cpu_frequency] not found, '
+                           'frequency data not available')
+            return None
+
+        if isinstance(entity, str) or isinstance(entity, unicode):
+            if not self._platform or 'clusters' not in self._platform:
+                self._log.warn('No platform data available. Cannot gather {} '
```

Comment:
Do we really need the cluster topology from the platforms file ? AFAICS we are still able to get the transition data with just a CPU id.

Reply:
Do you mean we should ask the user to always input a CPU ID? That would make the code a lot cleaner actually.

Reply:
Maybe you could go the same way as `plot{Cluster, CPU}Frequencies()`, i.e. have a method that expects a CPU ID (plots all by default) and another one that uses the platform data to do a pretty display with one plot per cluster.